tests/components/template/test_light.py

```diff
@@ -280,21 +711,38 @@ async def test_templatex_state_boolean(
 
 @pytest.mark.parametrize("count", [0])
 @pytest.mark.parametrize(
-    "light_config",
+    ("light_config", "style"),
     [
-        {
-            "test_template_light": {
-                **OPTIMISTIC_BRIGHTNESS_LIGHT_CONFIG,
-                "value_template": "{%- if false -%}",
-            }
-        },
-        {
-            "bad name here": {
+        (
+            {
+                "test_template_light": {
+                    **OPTIMISTIC_BRIGHTNESS_LIGHT_CONFIG,
+                    "value_template": "{%- if false -%}",
+                }
+            },
+            ConfigurationStyle.LEGACY,
+        ),
+        (
+            {
+                "bad name here": {
+                    **OPTIMISTIC_BRIGHTNESS_LIGHT_CONFIG,
+                    "value_template": "{{ 1== 1}}",
+                }
+            },
+            ConfigurationStyle.LEGACY,
+        ),
+        (
+            {"test_template_light": "Invalid"},
+            ConfigurationStyle.LEGACY,
+        ),
+        (
+            {
                 **OPTIMISTIC_BRIGHTNESS_LIGHT_CONFIG,
-                "value_template": "{{ 1== 1}}",
-            }
-        },
-        {"test_template_light": "Invalid"},
+                "name": "test_template_light",
+                "state": "{%- if false -%}",
+            },
+            ConfigurationStyle.MODERN,
+        ),
     ],
 )
 async def test_template_syntax_error(hass: HomeAssistant, setup_light) -> None:
```

Comment:
The name and docstring is misleading, the test tests config validation errors in general, not specifically template syntax errors